HSP2tools/readUCI.py

```diff
@@ -83,40 +85,64 @@ def fix_df(df, op, save, ddfaults, valid):
     if set(df.index) - valid:
         df = df.drop(index = set(df.index) - valid) # drop unnecessary ids
     for name1 in valid - set(df.index):
-        df = df.append(Series(name=name1))        # add missing ids with NaNs
-    if df.isna().any().any():                      # replace NaNs with defaults
+        df = pd.concat(
+            [df, pd.Series(name=name1)], axis="columns"
```

Comment:
wrong axis, btw. old append does "Append _rows_ of other to the end of caller," which is `df.concat(other, axis='rows')` or just `df.concat(other)`